src/store/card.js

```diff
@@ -86,8 +87,90 @@ export default {
 						return true
 					}
 
-					return card.title.toLowerCase().includes(getters.getSearchQuery.toLowerCase())
-						|| card.description.toLowerCase().includes(getters.getSearchQuery.toLowerCase())
+					let hasMatch = true
+					const matches = getters.getSearchQuery.match(/(?:[^\s"]+|"[^"]*")+/g)
+
+					const filterOutQuotes = (q) => {
+						if (q[0] === '"' && q[q.length - 1] === '"') {
+							return q.substr(1, -1)
+						}
+						return q
+					}
+					for (const match of matches) {
+						let [filter, query] = match.indexOf(':') !== -1 ? match.split(/:(.+)/) : [null, match]
+
+						if (filter === 'title') {
+							hasMatch = hasMatch && card.title.toLowerCase().includes(filterOutQuotes(query).toLowerCase())
+						} else if (filter === 'description') {
+							hasMatch = hasMatch && card.description.toLowerCase().includes(filterOutQuotes(query).toLowerCase())
+						} else if (filter === 'list') {
+							const stack = this.getters.stackById(card.stackId)
+							if (!stack) {
+								return false
+							}
+							hasMatch = hasMatch && stack.title.toLowerCase().includes(filterOutQuotes(query).toLowerCase())
+						} else if (filter === 'tag') {
+							hasMatch = hasMatch && card.labels.findIndex((label) => label.title.toLowerCase().includes(filterOutQuotes(query).toLowerCase())) !== -1
+						} else if (filter === 'date') {
+							const datediffHour = ((new Date(card.duedate) - new Date()) / 3600 / 1000)
+							query = filterOutQuotes(query)
+							switch (query) {
+							case 'overdue':
+								hasMatch = hasMatch && (card.overdue === 3)
+								break
+							case 'today':
+								hasMatch = hasMatch && (datediffHour > 0 && datediffHour <= 24 && card.duedate !== null)
+								break
+							case 'week':
+								hasMatch = hasMatch && (datediffHour > 0 && datediffHour <= 7 * 24 && card.duedate !== null)
+								break
+							case 'month':
+								hasMatch = hasMatch && (datediffHour > 0 && datediffHour <= 30 * 24 && card.duedate !== null)
+								break
+							case 'none':
+								hasMatch = hasMatch && (card.duedate === null)
+								break
+							}
+
+							if (card.duedate === null || !hasMatch) {
+								return false
+							}
+							const comparator = query[0] + (query[1] === '=' ? '=' : '')
+							const isValidComparator = ['<', '<=', '>', '>='].indexOf(comparator) !== -1
```

Comment:
throw error in case of no valid comparator ?

Reply:
Until now invalid queries would just attempt to perform an equal comparison for whatever is passed after the `:`. I'd keep that for now and move the handling improvement of that into a ticket so we can discuss where and how to show possible errors especially with the unified search, where we'd need server improvements to return possible hints or errors to the user.

Reply:
https://github.com/nextcloud/deck/issues/2973